caput/mpiarray.py

```diff
@@ -1036,6 +1036,177 @@ def _partition_io(self, skip=False, threshold=1.99):
         slices = [slice(start, end) for start, end in zip(starts, ends)]
         return split_axis, slices
 
+    @staticmethod
+    def _mpi_to_ndarray(inputs):
+        """Ensure a list with mixed MPIArrays and ndarrays are all ndarrays.
+
+        Additionally, ensure that all of the MPIArrays were are all distributed along the same axes.
+
+        Parameters
+        ----------
+        inputs : list of MPIArrays and ndarrays
+            All MPIArrays should be distributed along the same axes.
+
+        Returns
+        -------
+        args : list of ndarrays
+            The ndarrays are built from the local view of inputted MPIArrays.
+        dist_axis : int
+            The axis that all of the MPIArrays were distributed on.
+        """
+        args = []
+        dist_axis = None
+
+        for array in inputs:
+            if isinstance(array, MPIArray):
+                if dist_axis is None:
+                    dist_axis = array.axis
+                else:
+                    assert (
+                        dist_axis == array.axis
+                    ), "The distributed axis for all MPIArrays in an expression should be the same"
```

Comment:
Probably avoid using an `assert` which can be bypassed by run time options. Should explicitly raise a ValueError.